### PR TITLE
feat(automation): Add line numbers for repo scans for copyright and keyword scanners

### PR DESCRIPTION
**Pull request by rajuljha**
## Description
- Add line numbers for repo scans for copyright and keyword scanners

### Changes

- Add function to calculate line number for repo scans in `FormatResults.py`

## How to test

- Build the docker image from the `Dockerfile.ci` located at `utils/automation` directory.
- Set up a Github Actions workflow in any of your repositories with this [example](https://github.com/fossology/fossology/blob/master/utils/automation/.github-workflow.yml)
- Set up a Github Fine Grained Token ([link](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token))
- Set up a environment variable file and include the following Environment variables: 
```
GITHUB_ACTIONS=true
GITHUB_TOKEN=<your_github_token>
GITHUB_REPOSITORY=<your_gihtub_repo>
GITHUB_REPO_OWNER=<your_github_username>
GITHUB_REPO_URL=<youor_repo_url>
GITHUB_PULL_REQUEST=<your_pull_request_id>
```
- Run the docker container built with passing this environment variable file as a parameter
- Go to `/bin` directory and run the `fossologyscanner.py` script.

This PR is a superset of #2754 .
Please merge after #2754 

CC @GMishx @shaheemazmalmmd @avinal

### CODE REVIEW

File: utils/automation/FoScanner/FormatResults.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      """
+      formatted_diff = []
+      diff_lines = diff_content.splitlines()
+      left = right = 0
+      ll = rl = 0
+      for line in diff_lines:
+        match = re.match(r'^@@ -([0-9]+),([0-9]+) [+]([0-9]+),([0-9]+) @@', line)
+        if match:
+          left = int(match.group(1))
+          ll = len(match.group(2))
```
Comment (Kaushl2208) [inline utils/automation/FoScanner/FormatResults.py:36]: Maybe a bit more explanatory Variable name? 😁

**Review by Kaushl2208 — CHANGES_REQUESTED** [review]
Looks Good, Left a nitty picky comment (You can look into it for both the PRs)😮‍💨 !! 
Also, Maybe squash the commits

File: utils/automation/FoScanner/FormatResults.py (hunk 1 above)
Comment (rajuljha) [reply utils/automation/FoScanner/FormatResults.py]: Sure! Changed it!

**Review by avinal — CHANGES_REQUESTED** [review]
The rest looks good to me. Not critical, but if possible, limit the commit message to 50 chars and use the commit body to describe your changes further.

File: utils/automation/FoScanner/FormatResults.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2024 Rajul Jha <rajuljha49@gmail.com>
```
[the remark this answers is not in the document]
Comment (rajuljha) [reply utils/automation/FoScanner/FormatResults.py:3]: Okay! I have corrected it.

File: utils/automation/FoScanner/RepoSetup.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def get_diff_dir(self) -> str:
     else:
       changes = change_response
 
-    remove_diff_regex = re.compile(r"^([ +-])(.*)$", re.MULTILINE)
+    # remove_diff_regex = re.compile(r"^([ +-])(.*)$", re.MULTILINE)
```
[the remark this answers is not in the document]
Comment (rajuljha) [reply utils/automation/FoScanner/RepoSetup.py:114]: Nah! Forgot to remove the comment 😥

File: utils/automation/FoScanner/RepoSetup.py [hunk 4]
```diff
@@ -122,7 +122,8 @@ def get_diff_dir(self) -> str:
           if curr_dir != self.temp_dir.name:
             os.makedirs(name=curr_dir, exist_ok=True)
           curr_file = open(file=curr_file, mode='w+', encoding='UTF-8')
-          print(re.sub(remove_diff_regex, r"\2", change[change_key]),
-                file=curr_file)
+        #   print(re.sub(remove_diff_regex, r"\2", change[change_key]),
+        #         file=curr_file)
```
[the remark this answers is not in the document]
Comment (rajuljha) [reply utils/automation/FoScanner/RepoSetup.py:126]: Thanks for the review! I have made the changes 👍

**Comment by rajuljha** [comment]
This PR was raised as a superset of #2754 
During refactoring #2754 , the newer changes in this branch have been accustomed by #2754 
Hence, this PR will become redundant. We can close this after #2754 merges.

**Comment by rajuljha** [comment]
Closing this as mentioned in the comment above as #2754 is merged.

**Pull request closed without merging**